core/src/main/java/hudson/tasks/LogRotator.java

```diff
@@ -217,7 +216,10 @@ public void perform(Job<?,?> job) throws IOException, InterruptedException {
             //Collate all encountered exceptions into a single exception and throw that
             String msg = String.format(
                     "Failed to rotate logs for [%s]",
-                    Joiner.on(", ").join(exceptionMap.keySet())
+                    exceptionMap.keySet().stream().
+                        filter(run -> run!=null).
```

Comment:
Could use a method reference to `Objects::nonNull` rather than a custom lambda here:

```suggestion
                        filter(Objects::nonNull).
```

Reply:
I do not see much difference neither if we have any gain and it's even the same length :)